### DISCUSSION TITLE
Allow both server rendered and static generated routes

### DISCUSSION THREAD

**Discussion by flexdinesh** [withastro/roadmap#351] [Proposal]
I wanted to try Astro and ported my Next.js site and everything works great. And the site is now super duper fast. 🤯

One thing I would like is to have page specific control for server rendering and static generation.

> Astro will remain a static-site generator by default. But once you enable a server-side rendering adapter, every route in your pages directory becomes a server-rendered route and a few new features become available to you.

Once I added the vercel adapter all the routes are now server rendered. It'd nice to customise which pages are server rendered vs which pages are statically built. For example, I would like my homepage to be statically built and served from the edge and rest of the routes to be server rendered with some dynamic data fetching on the server.

It'd also be nice to have this same customisation for endpoints. Instead of setting a universal vercel target `edge` or `serverless` or `static`, I would like to be able to customise it per endpoint. 

Edge is great but limited. Libs like `@vercel/og` will only work in edge runtime. I would like one of my endpoints to be edge runtime but the rest to be serverless to use with node APIs.

Is there already a way to do this in the current version `^1.5.2`?

**Comment by delucis** [comment]
This is on our roadmap for this quarter! https://github.com/withastro/rfcs/discussions/302

**Reply by flexdinesh** [reply]
That's so awesome. Thank you!

**Comment by florian-lefebvre** [comment]
Closing because this is outdated
